src/character.cpp

```diff
@@ -5497,6 +5497,24 @@ int Character::weariness() const
     return weary.tracker - weary.intake * 0.5;
 }
 
+std::pair<int, int> Character::weariness_transition_progress() const
+{
+    // Mostly a duplicate of the below function. No real way to clean this up
+    int amount = weariness();
+    int threshold = weary_threshold();
+    amount -= threshold * get_option<float>( "WEARY_INITIAL_STEP" );
+    while( amount >= 0 ) {
+        amount -= threshold;
+        if( threshold > 20 ) {
+            threshold *= get_option<float>( "WEARY_THRESH_SCALING" );
+        }
+    }
+
+    // If we return the absolute value of the amount, it will work better
+    // Because as it decreases, we will approach a transition
+    return {std::abs( amount ), threshold};
```

Comment:
Does this mean that your transition bar goes down when resting _and_ when exerting?  I would expect the bar to fill all the way up and then you go to a better weariness level and a `:....` bar again.

Reply:
Here's how it works, thanks for reminding me to include this:
![weary](https://user-images.githubusercontent.com/42699974/103158761-5607d900-4776-11eb-853d-7ed54906e115.gif)
![weary2](https://user-images.githubusercontent.com/42699974/103158762-56a06f80-4776-11eb-89a6-5d0816aea104.gif)

Reply:
Cool animation, consider adding it to the first post!